source/javascripts/components/unified-editor/StepConfigDrawer/tabs/ConfigurationTab.tsx

```diff
@@ -17,40 +18,58 @@ function groupStepInputs(inputs?: StepInputVariable[]) {
 }
 
 const ConfigurationTab = () => {
-  const { data } = useStepDrawerContext();
-  const { mergedValues } = data ?? {};
-  const form = useFormContext<FormValues>();
+  const { data, workflowId, stepIndex } = useStepDrawerContext();
+  const updateStep = useDebounceCallback(useBitriseYmlStore(useShallow((s) => s.updateStep)), 250);
+  const updateStepInputs = useDebounceCallback(useBitriseYmlStore(useShallow((s) => s.updateStepInputs)), 250);
+
+  const mergedValues = data?.mergedValues ?? {};
+  const defaultValues = data?.defaultValues ?? {};
+
+  const onInputValueChange = (name: string, value?: string | null) => {
+    const clone = JSON.parse(JSON.stringify(mergedValues.inputs ?? [])) as StepInputVariable[];
+
+    clone.forEach(({ opts, ...input }, index) => {
```

Comment:
const index = clone.findIndex(({ opts, ...input }) => Object.keys(input).includes(name));

if (index !== -1) {
  // Update the value for the found index
  clone[index][name] = value;
}